v12/storage-dotnet-redundancy-ha-ra-grs/Program.cs

```diff
@@ -0,0 +1,262 @@
+﻿using System.Diagnostics.Tracing;
+using System.IO;
+using Azure;
+using Azure.Core;
+using Azure.Core.Diagnostics;
+using Azure.Core.Pipeline;
+using Azure.Identity;
+using Azure.Storage;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+namespace StorageRedundancy
+{
+    internal class ShowRequestInfoPolicy : HttpPipelinePolicy
+    {
+        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            string? host = message.Request.Uri.Host;
+            if (host is not null) 
+            {
+                string region = host.Contains("-secondary") ? "secondary" : "primary";
+                Console.WriteLine($"Request sent to {region} region endpoint");
+            }
+            
+            ProcessNext(message, pipeline);
+        }
+
+        public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            string? host = message.Request.Uri.Host;
+            
+            if (host is not null)
+            {
+                string region = host.Contains("-secondary") ? "secondary" : "primary";
+                Console.WriteLine($"Request sent to {region} region endpoint");
+            }
+            await ProcessNextAsync(message, pipeline);
+
+            
+        }
```

Comment:
```suggestion
        public override void Process(
            HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
        {
            ProcessMessage(message);
            ProcessNext(message, pipeline);
        }

        public override async ValueTask ProcessAsync(
            HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
        {
            ProcessMessage(message);            
            await ProcessNextAsync(message, pipeline);            
        }
        
        private void ProcessMessage(HttpMessage message)
        {
            string? host = message.Request.Uri.Host;            
            if (host is not null)
            {
                string region = host.Contains("-secondary") ? "secondary" : "primary";
                Console.WriteLine($"Request sent to {region} region endpoint");
            }
        }
```

Encapsulate shared logic, and format source (removing extra whitespace and new lines).